source/operation.ts

```diff
@@ -109,13 +112,13 @@ export function query(expression: string): QueryOperation {
  * @param  {string} expression API query expression
  * @return {Object}            API operation
  */
-export function search({
+export function search<TEntityType>({
```

Comment:
TEntityType is automatically inferred from the options type, so no need to specify it explicitly (please verify)

Reply:
That's clever if it works. Needs to be tested. The only issue I see is a lack of verification of the TEntitityType in the Options.